src/python/WMCore/ReqMgr/Service/Request.py

```diff
@@ -404,24 +405,56 @@ def _handleNoStatusUpdate(self, workload, request_args, dn):
-            # must update three places: GQ elements, workload_cache and workload spec
-            self.gq_service.updatePriority(workload.name(), request_args['RequestPriority'])
-            report = self.reqmgr_db_service.updateRequestProperty(workload.name(), request_args, dn)
-            workload.setPriority(request_args['RequestPriority'])
-            workload.saveCouchUrl(workload.specUrl())
-            cherrypy.log('Updated priority of "{}" to: {}'.format(workload.name(), request_args['RequestPriority']))
-        elif workqueue_stat_validation(request_args):
-            report = self.reqmgr_db_service.updateRequestStats(workload.name(), request_args)
-            cherrypy.log('Updated workqueue statistics of "{}", with:  {}'.format(workload.name(), request_args))
-        else:
-            msg = "There are invalid arguments for no-status update: %s" % request_args
-            raise InvalidSpecParameterValue(msg)
+        reqArgs = deepcopy(request_args)
+
+        if not reqArgs:
+            cherrypy.log(f"Nothing to be changed at this stage for {workload.name()}")
+            return 'OK'
+
+        if workqueue_stat_validation(reqArgs):
+            report = self.reqmgr_db_service.updateRequestStats(workload.name(), reqArgs)
+            cherrypy.log('Updated workqueue statistics of "{}", with:  {}'.format(workload.name(), reqArgs))
+            return report
+
+        reqArgsNothandled = []
+        for reqArg in reqArgs:
+            if reqArg == 'RequestPriority':
+                validate_request_priority(reqArgs)
+                # must update three places: GQ elements, workload_cache and workload spec
+                self.gq_service.updatePriority(workload.name(), reqArgs['RequestPriority'])
+                workload.setPriority(reqArgs['RequestPriority'])
+                cherrypy.log('Updated priority of "{}" to: {}'.format(workload.name(), reqArgs['RequestPriority']))
+            elif reqArg == "SiteWhitelist":
+                workload.setSiteWhitelist(reqArgs["SiteWhitelist"])
+                cherrypy.log('Updated SiteWhitelist of "{}", with:  {}'.format(workload.name(), reqArgs['SiteWhitelist']))
+            elif reqArg == "SiteBlacklist":
+                workload.setSiteBlacklist(reqArgs["SiteBlacklist"])
+                cherrypy.log('Updated SiteBlacklist of "{}", with:  {}'.format(workload.name(), reqArgs['SiteBlacklist']))
+            else:
+                reqArgsNothandled.append(reqArg)
+                cherrypy.log("Unhandled argument for no-status update: %s" % reqArg)
+
+        reqStatus = self.reqmgr_db_service.getRequestByNames(workload.name())[workload.name()]['RequestStatus']
+        cherrypy.log(f"CurrentRequest status: {reqStatus}")
+        if reqArgsNothandled:
+            if reqStatus == 'assignment-approved':
+                cherrypy.log(f"Handling assignment-approved arguments differently!")
+                self._handleAssignmentStateTransition(workload, request_args, dn)
```

Comment:
The call to `_handleAssignmentStateTransition` should only be performed when we are making a workflow assignment, which is defined from a state transition from `assignment-approved` to `assigned`, which we do not have in here. What am I missing here?

Reply:
And this exactly is the line  which solves  the bug because of which this PR was stopped. And in relation also to my previous comment - as long as we do not want to refactor the whole logic how we make those request parameters changes we will still have to treat `assignment-approved` status separately even when it comes to `non-state` transition changes. 
the alternative would be to re-implement all the calls from the method  `_handleAssignmentStateTransition` (basically copy paste them) here as well.